email-smtp/src/main/java/io/commercetools/sunrise/email/smtp/SmtpAuthEmailSender.java

```diff
@@ -1,27 +1,242 @@
+     * sender waits to be connected to the SMTP server, waits to read data from the SMTP server, or waits to write data
+     * to the SMTP server, an {@link EmailSenderException} will be raised and sending of the e-mail that caused the
+     * issue will be aborted. If this happens before the e-mail has been received by the SMTP server, the e-mail will
+     * not be sent, as users of this service will not typically implement error-handling in such cases. In general,
+     * shorter timeouts may lead to more messages being lost in the abscence of further error handling, longer timeouts
+     * may lead to more tasks being queued if a {@link ForkJoinPool} is used as {@link Executor}; run-time monitoring of
+     * the {@link ForkJoinPool} might help choose a suitable timeout value.
+     * <h1>Message size</h1>
+     * The e-mail sender does not limit the size of the messages it accepts. Because the {@link #send(MessageEditor)}
+     * method fills messages before sending is attempted within the {@link Executor}, it is theoretically possible that
+     * a large queue of messages waiting to be send consumes all available memory. Users of this API should therefore
+     * ensure that messages do not exceed a certain size.
+     * <h1>Configuration correctness</h1>
+     * Note that this constructor does not fail fast: the constructor does not create a connection to the mail server to
+     * ensure that the connection details and credentials are correct. If you would like to ensure a correct
+     * configuration, send a test message at startup and decide on your own what to do if that fails.
+     * <h1>Shutdown</h1>
+     * If the JVM running this e-mail sender is shut down, the {@link Executor} passed to this constructor should be
+     * shut down, too, in a way that ensures that all messages submitted to {@link #send(MessageEditor)} have been sent
+     * via SMTP - or a timeout expires in the case too many messages are waiting. If a {@link ForkJoinPool} is used as
+     * {@link Executor}, {@link ForkJoinPool#awaitQuiescence(long, TimeUnit)} may be used for this purpose.
+     *
+     * @param executor          the executor to use, e.g. {@link ForkJoinPool#commonPool()} may be used, but see above
+     * @param host              the name of the host running the SMTP server to use
+     * @param port              the port of the SMTP server to use
+     * @param transportSecurity how to secure the SMTP connection
+     * @param username          the username to use for authenticating with the SMTP server
+     * @param password          the password to authenticate with
+     * @param timeoutMs         the timeout for creating, reading from and writing to SMTP connections in
+     *                          milliseconds, see above for details
+     */
+    public SmtpAuthEmailSender(@Nonnull final Executor executor, @Nonnull final String host, final int port,
```

Comment:
Focusing on the constructor (so I'm not touching the same topic as @schleichardt , which was aimed for subclasses, instead I'm just a user of the `SmtpAuthEmailSender`) we are mixing two type of parameters:
- SMTP related parameters (i.e. `host`, `port`, `transportSecurity`, `username`, `password`), which might also grow in the future, and
- Fine-tuning parameters, such as `executor` and `timeoutMs`, which for the getting started might be a bit overwhelming.
I would try to organize them differently. So for the first case I would use a Class to encapsulate the proper SMTP configuration, which might follow the Builder pattern to enable expansion in the future. The reason why I would not use the `Properties` instance is because I don't want developers to know the property key they have to use e.g. for host `mail.smtp.host `. Of course internally we will use the Properties class and with the `createProperties` and `properties` methods, but not from outside.
For the second topic, if we use a config class for the rest, then these two factors (timeout and executor) can go at the end together. I would judge if we provide a default constructor with these two values already defined for the fast getting started situation, some values that can generally give an acceptable result before the fine-tuning.
I hope I explained myself properly, what do you think @schleichardt @selo-ecube ?